app/static/js/createEvents.js

```diff
@@ -60,18 +60,86 @@ function format24to12HourTime(timeStr){
       });
 
   }
+
+
+function getSelectedFiles(){
+  let _fileHolder = new DataTransfer();
+  $(".fileHolder").each(function(){
+    _fileHolder.items.add($(this).data("file"));
+  });
+  return _fileHolder.files;
+}
+
+
+function hasUniqueFileName(fileName){
+  let nameHolders = $(".fileName").toArray();
+  for (let i = 0; i < nameHolders.length; i++){
+    if (fileName == $(nameHolders[i]).data("filename")) return false;
+  }
+  return true;
+}
+
+
+
 /*
  * Run when the webpage is ready for javascript
  */
 $(document).ready(function() {
   if ( $("#startDatePicker").val() != $("#endDatePicker").val()){
-
     calculateRecurringEventFrequency();
   }
 
-    $("#attachmentObject").fileinput({
-        allowedFileExtensions:["pdf","jpg","png","gif", "csv", "docx", "jpg", "jpeg", "jfif"]
-    })
+    var fileNum = 0;
+    $("#attachmentObject").on('change', function() {
+      const selectedFiles = $("#attachmentObject").prop('files');
+      for (let i = 0; i < selectedFiles.length; i++) {
+        const file = selectedFiles[i];
+        if (hasUniqueFileName(file.name)){
+          let fileName = (file.name.length > 25) ? file.name.slice(0,10) + '...' + file.name.slice(-10) : file.name;
+          let fileExtension = file.name.split(".").pop();
+          let iconClass = '';
+          switch(fileExtension) {
+            case 'jpg': 
+            case 'png':
+            case 'jpeg':
+              iconClass = "bi-file-image";
+              break
+            case 'pdf':
+              iconClass = 'bi-filetype-pdf';
+              break
+            case 'docx':
+              iconClass = 'bi-filetype-docx';
+              break
+            case 'xlsx':
+              iconClass = 'bi-filetype-xlsx';
+              break
+            default:
+              iconClass = 'bi-file-earmark-arrow-up';
+          }
+          $("#attachedObjectContainer").append("<div class='border row p-0 m-0' id='attachedFilesRow" +fileNum+"'></div>")
+          $("#attachedFilesRow"+fileNum).append("<i class='col-auto fs-3 px-3 bi " + iconClass + "'></i>")
+          $("#attachedFilesRow"+fileNum).append("<div id='attachedFile" + fileNum + "' data-filename='" + file.name + "' class='fileName col-auto pt-2'>" + fileName + "</div>");
+          $("#attachedFilesRow"+fileNum).append("<div class='col' style='text-align:right'> \
+                                                  <div class='btn btn-danger fileHolder p-1 my-1 mx-1' id='trash" + fileNum + "'>\
+                                                    <span class='bi bi-trash fs-6'></span>\
+                                                  </div>\
+                                                </div>")
```

Comment:
Since you are already doing a big block of html, you could combine these append statements into one html block

Reply:
I'd also suggest creating a variable for the id so you don't have to repeat `"#attachedFilesRow"+fileNum` in so many places.

Reply:
Since we've taken your initial suggestion, we no longer have the need to create a variable for the id! All append statements have been collapsed into a single append statement with a multiline string.